content/academy/pro_apify_scraper_development/integrating_webhooks.md

```diff
@@ -0,0 +1,184 @@
+    if (!acc[curr.asin] || prevPrive > price) acc[curr.asin] = curr;
+
+    // Return the map
+    return acc;
+}, {});
+```
+
+The results should be an array, so finally, we can take the map we just created and push an array of all of its values to the actor's default dataset:
+
+```JavaScript
+await Apify.pushData(Object.values(filtered));
+```
+
+Our final code looks like this:
+
+```JavaScript
+const Apify = require('apify');
+
+Apify.main(async () => {
+    const { datasetId } = await Apify.getInput();
+    const dataset = await Apify.openDataset(datasetId);
+
+    const { items } = await dataset.getData();
+
+    const filtered = items.reduce((acc, curr) => {
+        const prevPrive = acc?.[curr.asin] ? +acc[curr.asin].offer.slice(1) : null;
+        const price = +curr.offer.slice(1);
+
+        if (!acc[curr.asin] || prevPrive > price) acc[curr.asin] = curr;
+
+        return acc;
+    }, {});
+
+    await Apify.pushData(Object.values(filtered));
+});
+```
+
+Cool! But **wait**, don't forget to configure the **INPUT_SCHEMA.json** file as well! It's not necessary to do this step, as we'll be calling the actor through Apify's API within a webhook, but it's still good to get into the habit of writing quality input schemas that describe the input values your actors are expecting.
+
+```JSON
+{
+    "title": "Amazon Filter Actor",
+    "type": "object",
+    "schemaVersion": 1,
+    "properties": {
+        "datasetId": {
+            "title": "Dataset ID",
+            "type": "string",
+            "description": "Enter the ID of the dataset.",
+            "editor": "textfield"
+        }
+    },
+    "required": ["datasetId"]
+}
+```
+
+Now we're done, and we can push it up to the Apify platform with the `apify push` command.
+
+## [](#setting-up-the-webhook) Setting up the webhook
+
+Since we'll be calling the actor via the [Apify API](https://docs.apify.com/tutorials/integrations/run-actor-and-retrieve-data-via-api#run-an-actor-or-task), we'll need to grab hold of the ID of the actor we just created and pushed to the platform. The ID is always accessible through the **Settings** page of the actor.
+
+![Actor ID in actor settings]({{@asset pro_apify_scraper_development/images/actor-settings.webp}})
+
+With this `actorId`, and our `token`, which is retrievable through **Settings > Integrations** on the Apify Console, we can construct a link which will call the actor:
+
+```text
+https://api.apify.com/v2/acts/Yk1bieximsduYDydP/runs?token=YOUR_TOKEN_HERE
+```
+
+We can also use our username and the name of the actor like this:
+
+```text
+https://api.apify.com/v2/acts/USERNAME~filter-actor/runs?token=YOUR_TOKEN_HERE
+```
+
+Whichever one you choose is totally up to preference.
+
+Next, within the actor, we will click the **Integrations** tab and choose **Webhook**, then fill out the details to look like this:
+
+![Configuring a webhook]({{@asset pro_apify_scraper_development/images/adding-webhook.webp}})
```

Comment:
Please trim off the bottom bit of the image where the Intercom button and the bottom of the box show - it will be more slick that way
<img width="755" alt="Screenshot 2022-04-13 at 10 46 58" src="https://user-images.githubusercontent.com/38246758/163137773-cfcd5962-cc18-4da0-9d94-78160d35479d.png">